benchcheck.go

```diff
@@ -99,3 +133,65 @@ func RunBench(mod Module) ([]string, error) {
 	}
 	return results, nil
 }
+
+// Stat compares two benchmark results providing a set of results.
+// oldres and newres must be multiple strings where each line follows
+// Go's benchmark output format.
+func Stat(oldres []string, newres []string) ([]StatResult, error) {
+	// We are using benchstat defaults:
+	//	- https://cs.opensource.google/go/x/perf/+/master:cmd/benchstat/main.go;l=117
+	const (
+		alpha   = 0.05
+		geomean = false
+	)
+	c := &benchstat.Collection{
+		Alpha:      alpha,
+		AddGeoMean: geomean,
+		DeltaTest:  benchstat.UTest,
+	}
+	if err := c.AddFile("old", resultsReader(oldres)); err != nil {
+		return nil, fmt.Errorf("parsing old results: %v", err)
+	}
+	if err := c.AddFile("new", resultsReader(newres)); err != nil {
+		return nil, fmt.Errorf("parsing new results: %v", err)
+	}
+	return newStatResults(c.Tables()), nil
+}
+
+func newStatResults(tables []*benchstat.Table) []StatResult {
+	res := make([]StatResult, len(tables))
+
+	for i, table := range tables {
+		res[i] = StatResult{
+			Metric:       table.Metric,
+			BenchResults: newBenchResults(table.Rows),
+		}
+	}
+
+	return res
+}
+
+func newBenchResults(rows []*benchstat.Row) []BenchResult {
+	res := make([]BenchResult, len(rows))
+
+	for i, row := range rows {
+		if len(row.Metrics) != 2 {
```

Comment:
is this true even in the case that `new` has benchmarks not present in the `old`?
Eg.: the newer commit introduced a new Bench* test.

Reply:
asking because this is quite common.

Reply:
will test it

Reply:
Addes some tests validating that this is true (in the sense that it doesn't panic) when there are a few functions that are not present on both, when one of them is empty or when they have benchmarkes but they don't match. all result in empty stats.